pom.xml

```diff
@@ -99,11 +99,18 @@
 
     <properties>
         <!--  Java EE specs  -->
-        <cdi-version>2.0</cdi-version>
-        <jaxrs-version>2.1</jaxrs-version>
-        <jsonb-version>1.0</jsonb-version>
-        <jsonp-version>1.1</jsonp-version>
-        <annotation-version>1.3</annotation-version>
+        <javax-cdi-version>2.0</javax-cdi-version>
+        <javax-jaxrs-version>2.1</javax-jaxrs-version>
+        <javax-jsonb-version>1.0</javax-jsonb-version>
+        <javax-jsonp-version>1.1</javax-jsonp-version>
+        <javax-annotation-version>1.3</javax-annotation-version>
+
+        <!--  Jakarta EE specs  -->
+        <jakarta-cdi-version>[2.0,2.1)</jakarta-cdi-version>
```

Comment:
Version ranges are quite problematic on maven.
Example:

Lets say we use this range `[2.0,2.1)`
And there's a version `2.0.199` in the repository system.
Your build will use that. yay!

But lets say the project releases both versions `2.1.0-alpha1`, and `2.0.200`.
You now do a fresh build and get ... `2.1.0-alpha1`.
Why? well, that is technically below the upper bound of `2.1` (it's not a stable release yet)

And if the project releases a final `2.1.0` (stable release) then you will not get that version, as it's stable and satisfies the range you specified.

Generally speaking, all maven projects that have used version ranges eventually stop using them as they have surprising/unexpected behaviors.

Reply:
Good point well made @joakime - and good mvn-fu, I learn many things every day.

One of the reasons why in https://github.com/eclipse/microprofile/pull/132/commits/82916b33a654f0d8cd4ada3a7bc1c28554953db0 we removed their use for
any specs that were not controlled by MicroProfile in #132 was for these sorts of
concerns. We only planned to use ranges when we control both sides of the link
but you might be right on #132 so I will add a link here.

I will have a further play with attempting to find a lower upper bound for 2.1) too - thanks
again for the intresting info - you may change the outcome of #132 so quite timely too :-)

Reply:
Hi @joakime, in order to keep this PR focusing on illustrating the ability to give users choice in their Java/Jakarta EE8 transitive dependencies for microprofile specs - I have removed the ranges in the Jakarta dependencies in this PR in 7970444. I look further into use of ranges in #132 so your point is still a very useful one that I am looking into. Thanks again.

Reply:
Hi @joakime to paraphrase the above I saw/understood:

"Lets say we use this range [2.0,2.1)
...the project releases both versions 2.1.0-alpha1, and 2.0.200.
You now do a fresh build and get ... 2.1.0-alpha1."

This is, as I am sure you are aware, because 2.1 is 'expanded' to 2.1.0
for comparison and strings such as
(all uppercase OR lowercase and prefixed or not with a "-")
alpha, beta, rc, snapshot are all considered 'special' and 'earlier' than the
release they decorate. 

E.g. 

So with 
```shell
$ export MAVEN_HOME=/Users/hutchig/apache-maven-3.6.1 && export CLASSPATH=$CLASSPATH:$MAVEN_HOME/lib/* && java org.apache.maven.artifact.versioning.ComparableVersion "[2.0,2.1)" 2.1.0-alpha1
Display parameters as parsed by Maven (in canonical form) and comparison result:
1. [2.0,2.1) == [-2-,-2.1-)
   [2.0,2.1) < 2.1.0-alpha1
2. 2.1.0-alpha1 == 2.1-alpha-1
```
The above is not really helpful (is there a better range tester?), what is illustrative is: 

```shell
$ export MAVEN_HOME=/Users/hutchig/apache-maven-3.6.1 && export CLASSPATH=$CLASSPATH:$MAVEN_HOME/lib/* && java org.apache.maven.artifact.versioning.ComparableVersion  2.1-alpha 2.1
Display parameters as parsed by Maven (in canonical form) and comparison result:
1. 2.1-alpha == 2.1-alpha
   2.1-alpha < 2.1
2. 2.1 == 2.1
~
$ export MAVEN_HOME=/Users/hutchig/apache-maven-3.6.1 && export CLASSPATH=$CLASSPATH:$MAVEN_HOME/lib/* && java org.apache.maven.artifact.versioning.ComparableVersion  2.1-aaaaa 2.1
Display parameters as parsed by Maven (in canonical form) and comparison result:
1. 2.1-aaaaa == 2.1-aaaaa
   2.1-aaaaa > 2.1
2. 2.1 == 2.1
~
$ export MAVEN_HOME=/Users/hutchig/apache-maven-3.6.1 && export CLASSPATH=$CLASSPATH:$MAVEN_HOME/lib/* && java org.apache.maven.artifact.versioning.ComparableVersion  2.1-aaaaa 2.1-alpha
Display parameters as parsed by Maven (in canonical form) and comparison result:
1. 2.1-aaaaa == 2.1-aaaaa
   2.1-aaaaa > 2.1-alpha
2. 2.1-alpha == 2.1-alpha
$ export MAVEN_HOME=/Users/hutchig/apache-maven-3.6.1 && export CLASSPATH=$CLASSPATH:$MAVEN_HOME/lib/* && java org.apache.maven.artifact.versioning.ComparableVersion  2.1-alpha  2.1alpha
Display parameters as parsed by Maven (in canonical form) and comparison result:
1. 2.1-alpha == 2.1-alpha
   2.1-alpha == 2.1alpha
2. 2.1alpha == 2.1-alpha
```
After some research [1.1,1.2alpha) is the correct term.
( or [1.1,1.2-alpha) ) to constrain a version (water) tightly
to 1.1* as "alpha" is the true lower bound decorator.

Reply:
I will update #132.